src/permissions/PermissionController.test.ts

```diff
@@ -3999,6 +3964,58 @@ describe('PermissionController', () => {
     });
   });
 
+  // If permission is already accepted the method should not throw error and also should not call ApprovalController:acceptRequest
+  // This is to make the method idempotent and multiple calls with same parameters will
+  // leave controller in same state and function will not throw error.
+  describe('acceptPermissionsRequest (idempotent tests)', () => {
+    it('does not throw error if the request does not exist', async () => {
+      const options = getPermissionControllerOptions();
+      const origin = 'metamask.io';
+      const id = 'foobar';
+
+      const controller = getDefaultPermissionController(options);
+
+      expect(async () => {
+        await controller.acceptPermissionsRequest({
```

Comment:
When you're testing for something not to throw then you can the `await` away. Otherwise, this will throw at this spot and the assertion will never get run.